README.md

```diff
@@ -0,0 +1,390 @@
+mapper]*).  None of this is useful on its own, so we also provide parameters to
+install user mapping data and CA certificates.
+
+To associate the certificates contained on a smart card or other PKCS#11 token,
+use the `digest_mappings` parameter like so:
+
+```puppet
+class { 'pam_pkcs11':
+  digest_mappings => {
+    'alice' => '79:E7:27:38:59:24:C6:AD:92:E5:AA:FA:20:0F:D6:9A:D5:47:87:DF',
+    'bob'   => 'DD:75:AD:96:0E:CD:BD:25:E7:27:02:8B:34:3D:E4:08:FA:44:A8:31',
+  },
+}
+```
+
+Handling CA certificates is more difficult because depending on the
+cryptographic backend they are stored differently.  The most common back-end
+options are NSS and OpenSSL.  Unfortunately, this module does not currently
+support NSS database storage.  See the [Limitations section](#limitations)
+below for more details.
+
+For Gentoo and Debian-based systems, the parameters `ca_dir_source` and
+`ca_dir_sourceselect` provide a way to specify CA file sources for the OpenSSL
+hash dir back-end.  The parameters are simply passed on to a puppet file
+resource, as `source` and `sourceselect` with the caveat that `ca_source_dir`
+has to be an array of strings.  See the puppet documentation about the
+[file](https://puppet.com/docs/puppet/7/types/file.html) resource for more
+information.
+
+Extending the example from before, here all the CA certificates are stored in the
+`ca_dir` sub-directory of a "*files*" module are installed on the target host:
+
+```puppet
+class { 'pam_pkcs11':
+  digest_mappings => {
+    'alice' => '79:E7:27:38:59:24:C6:AD:92:E5:AA:FA:20:0F:D6:9A:D5:47:87:DF',
+    'bob'   => 'DD:75:AD:96:0E:CD:BD:25:E7:27:02:8B:34:3D:E4:08:FA:44:A8:31',
+  },
+  ca_dir_source => ['puppet:///modules/files/ca_dir'],
+}
+```
+
+With this configuration, Alice and Bob should be able to authenticate, provided
+the certificates on their smart cards were issued by a trusted CA and have not
+been revoked.
+
+[digest mapper]: http://opensc.github.io/pam_pkcs11/doc/pam_pkcs11.html#idp5267200
+
+## Usage
+
+The base `pam_pkcs11` class is the *primary point of entry* for this module.
+This class should be declared with its parameters defined as appropriate to the
+deployment environment.
+
+The sub-class `pam_pkcs11::pkcs11_eventmgr` provides a supplementary interface
+to configure the `pkcs11_eventmgr(1)` session lock daemon.  It will be enabled
+via the `pam_pkcs11::manage_pkcs11_eventmgr` parameter.
+
+## Reference
+
+### Classes
+
+#### Public Classes
+
+* [`pam_pkcs11`](#::pam_pkcs11): The primary point of entry to this module.
+* [`pam_pkcs11::pkcs11_eventmgr`](#::pam_pkcs11::pkcs11_eventmgr):
+  Manages `pkcs11_eventmgr(1)`, the session lock helper daemon.
+
+#### Private Classes
+
+* [`pam_pkcs11::install`]: Manages installation of the *pam_pkcs11* package.
+* [`pam_pkcs11::config`]: Manages configuration of the various configuration files.
+* [`pam_pkcs11::params`]: Contains the default configuration parameters.
```

Comment:
What are the links here supposed to lead to?

Reply:
Removed the links. And fixed the links in the 'Public Classes' section.